internal/functions/efgs/keys-operations_test.go

```diff
@@ -0,0 +1,233 @@
+					{Key: "d", IntervalNumber: 10, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "e", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "f", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "g", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "h", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				batchSize:      50,
+				maxOverlapping: 2,
+			},
+			wantChunks: [][]ExpKey{
+				{
+					{Key: "b", IntervalNumber: 0, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "g", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				{
+					{Key: "e", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "f", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "d", IntervalNumber: 10, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				{
+					{Key: "h", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "c", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "a", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+				},
+			},
+			wantMatchResult: true,
+			wantChunksCount: 3,
+		},
+		{
+			name: "efgsDownloadedKeys",
+			args: args{
+				keys:           loadTestingKeys("aligned"),
+				batchSize:      50,
+				maxOverlapping: 15,
+			},
+			wantChunks:      nil,
+			wantMatchResult: false,
+			wantChunksCount: 101,
+		},
+		{
+			name: "efgsDownloadedKeysUnaligned",
+			args: args{
+				keys:           loadTestingKeys("unaligned"),
+				batchSize:      50,
+				maxOverlapping: 15,
+			},
+			wantChunks:      nil,
+			wantMatchResult: false,
+			wantChunksCount: 132,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotChunks := splitKeys(tt.args.keys, tt.args.batchSize, tt.args.maxOverlapping)
+
+			if len(gotChunks) != tt.wantChunksCount {
+				t.Errorf("splitKeys() check count error; got %v, want %v", len(gotChunks), tt.wantChunksCount)
+				return
+			}
+
+			for _, chunk := range gotChunks {
+				if err := checkKeyServerBatch(chunk, tt.args.batchSize, tt.args.maxOverlapping); err != nil {
+					t.Errorf("splitKeys() check error = %+v", err)
+					return
+				}
+			}
+
+			t.Logf("Chunks: %v", len(gotChunks))
+
+			if tt.wantMatchResult && !reflect.DeepEqual(gotChunks, tt.wantChunks) {
+				t.Errorf("splitKeys() gotChunks = %+v, want %+v", gotChunks, tt.wantChunks)
+			}
+		})
+	}
+}
+
+func loadTestingKeys(suffix string) (efgsDownloadedKeys []ExpKey) {
+	bytes, err := ioutil.ReadFile(fmt.Sprintf("../../../testdata/efgs-downloaded-keys-%v.json", suffix))
+	if err != nil {
+		panic(err)
+	}
+
+	if err = json.Unmarshal(bytes, &efgsDownloadedKeys); err != nil {
+		panic(err)
+	}
+
+	return efgsDownloadedKeys
+}
+
+// This is more-or-less copied from https://github.com/google/exposure-notifications-server/blob/release-0.9/internal/publish/model/exposure_model.go#L418
```

Comment:
should we ask them to move it from `internal` to `api` and reuse it as a lib?

Reply:
The original code is much more complex but yeah, we can try. However I think we're the only ones who would benefit from that.